### PR TITLE
Post stuff

### PR DESCRIPTION
**Pull request by BogdanCosminBarbu**

### CODE REVIEW

**Review by paraschevanegru — APPROVED** [review]

File: Controllers/PostController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using Interfaces;
+
+namespace Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : ControllerBase
+    {
+        private static IPost _postService = new PostService();
+        private static IComment _commentService = new CommentService();
```
Comment (alexcogojocaru) [inline Controllers/PostController.cs]: pls remove unused services

File: Controllers/PostController.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            PostEntity post = _postService.GetPost(id);
+
+            return (post != null) ? Ok(post) : NotFound();
+        }
+
+        [HttpPost]
+        public IActionResult PostPost([FromBody] PostEntity post)
+        {
+            if ((post == null) || (_postService.GetPost(post.Id) == null))
```
Comment (alexcogojocaru) [inline Controllers/PostController.cs]: here it should be only post == null

**Review by alexcogojocaru — CHANGES_REQUESTED** [review]

**Comment by alexcogojocaru** [comment]
also check why the get request doesn't work

File: Models/PostEntity.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{
+    public class PostEntity
+    {
+        [JsonProperty("id")]
+        public uint Id { get; set; }
+
+        [JsonProperty("username")]
+        public string Username { get; set; }
+
+        [JsonProperty("content")]
```
Comment (alexcogojocaru) [inline Models/PostEntity.cs:30]: why content?

**Review by alexcogojocaru — CHANGES_REQUESTED** [review]

File: Controllers/PostController.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+using Microsoft.AspNetCore.Mvc;
```
Comment (alexcogojocaru) [inline Controllers/PostController.cs:17]: add a header

File: Interfaces/IPost.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+/**************************************************************************
+ *                                                                        *
+ *  File:        IComment.cs                                              *
+ *  Copyright:   (c) 2021, Cojocaru Daniel-Alexandru                      *
```
Comment (alexcogojocaru) [inline Interfaces/IPost.cs]: i didn't write this

File: Models/PostEntity.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+/**************************************************************************
+ *                                                                        *
+ *  File:        CommentEntity.cs                                         *
+ *  Copyright:   (c) 2021, Cojocaru Daniel-Alexandru                      *
```
Comment (alexcogojocaru) [inline Models/PostEntity.cs]: or this

File: Services/PostService.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
+/**************************************************************************
+ *                                                                        *
+ *  File:        CommentService.cs                                        *
+ *  Copyright:   (c) 2021, Cojocaru Daniel-Alexandru                      *
```
Comment (alexcogojocaru) [inline Services/PostService.cs]: pls add your name :(

File: UnitTests/PostServiceUnitTest.cs [hunk 8]
```diff
@@ -0,0 +1,237 @@
+/**************************************************************************
+ *                                                                        *
+ *  File:        CommentServiceUnitTest.cs                                *
+ *  Copyright:   (c) 2021, Cojocaru Daniel-Alexandru                      *
```
Comment (alexcogojocaru) [inline UnitTests/PostServiceUnitTest.cs]: and here 🥇

**Review by alexcogojocaru — CHANGES_REQUESTED** [review]

**Review by alexcogojocaru — APPROVED** [review]

**Pull request merged**